.github/workflows/release.yaml

```diff
@@ -0,0 +1,34 @@
+name: Build and Push Docker image
+
+on:
+  release:
+    types: [published]
```

Comment:
I'd like to build on all PRs, but only push on merges to main - I think this means it'll  only build when releases are published?

Reply:
oh, actually we already do build on PRs - https://github.com/grafana/smtprelay/blob/1a6e558df42ce7c96ffaa960d8b1a4b606592a5c/.github/workflows/docker.yml

so maybe this is OK... 🤔

Reply:
I guess it should tag and push the image when we do a new release, and if we push to main it should build and push with the `latest` tag